File: 401~500/404.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

var sumOfLeftLeaves = function(root,t=0) {
    sum=0;
    if(root===null) return 0;
    console.log(root.val);
    if(root.left===null && root.right===null && t!==0) {
        return root.val;
    }else if(root.left===null){
        sum+=sumOfLeftLeaves(root.right);
    }else if(root.right===null){
        sum+=sumOfLeftLeaves(root.left,1);
    }else{
        sum+=sumOfLeftLeaves(root.right);
        sum+=sumOfLeftLeaves(root.left,1);
    }
    return sum;
};

File: 401~500/404.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumOfLeftLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        count = [0]
        self.helper(root, count)
        return count[0]

    def helper(self, root, count):
        if root:
            if not root.left and not root.right:
                return root.val
            else:
                left_leaf = self.helper(root.left, count)
                count[0] += left_leaf if left_leaf else 0
                self.helper(root.right, count)
